adafruit_bitbangio.py

```diff
@@ -67,10 +68,15 @@ def _check_lock(self) -> Literal[True]:
             raise RuntimeError("First call try_lock()")
         return True
 
-    def __enter__(self):
+    def __enter__(self) -> "_BitBangIO":
```

Comment:
Yup, you got it!  It's a forward reference of itself, the underscore is irrelevant to typing - it just signify to people importing this file that `_BitBangIO` isn't intended to be used directly (it's a convention of naming classes).  Essentially, you want people importing classes like `I2C` here and not the basis class `_BitBangIO` that isn't intended for direct use itself.